app/src/main/java/com/amaze/filemanager/utils/Utils.java

```diff
@@ -38,6 +39,7 @@
     private static final String INPUT_INTENT_BLACKLIST_PIPE = "\\|";
     private static final String INPUT_INTENT_BLACKLIST_AMP = "&&";
     private static final String INPUT_INTENT_BLACKLIST_DOTS = "\\.\\.\\.";
+    private static final Pattern FILENAME_REGEX = Pattern.compile("[\\\\\\/:\\*\\?\"<>\\|\\x01-\\x1F\\x7F]", Pattern.CASE_INSENSITIVE);
```

Comment:
Off topic: I can't really read this RegEx, but it seems to be filtering [more than it should](https://unix.stackexchange.com/a/230299/274342). Not sure how Android treats valid filenames though.

Reply:
That regexp is for Windows compatibility. Even if Linux filesystems allow for most of those symbols, such files can't be copied to FAT-formatted SD card — Windows filesystem drivers won't able to interact with them.

Reply:
Very well